samples-test/samples/JavaScript/count_expression.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Parses and evaluates a mathematical expression string
+ * @param {string} expression - The mathematical expression to evaluate
+ * @returns {number} The result of the expression
+ */
+function parseExpression(expression) {
+    try {
+        // Remove whitespace and validate input
+        const cleanExpression = expression.replace(/\s+/g, '');
+        
+        // Check for invalid characters
+        if (!/^[0-9+\-*/().]+$/.test(cleanExpression)) {

```

Comment:
I know this is just a sample, but it's probably better to check if `invalidRegex.test` returns `true` rather than `validRegex.test` returns `false`.

`if (/[^0-9+\-*/()\.]/.test(cleanExpression) { // is invalid`.

This regex stops at the first invalid character, rather than checking *every* character.

Again, just a sample so you don't need to fix this, just wanted to point this out 🙂